src/cargo_ops/elaborate_workspace.rs

```diff
@@ -301,4 +319,102 @@ impl<'ela> ElaborateWorkspace<'ela> {
 
         Ok(lines.len() as i32)
     }
+
+    pub fn print_json(
+        &'ela self,
+        options: &Options,
+        root: &'ela PackageId,
+        preceding_line: bool,
+    ) -> CargoResult<i32> {
+        let mut crate_graph = CrateMetadata {
+            dependencies: HashSet::new(),
+        };
+        let mut queue = VecDeque::new();
+        queue.push_back(vec![root]);
+        
+        while let Some(path) = queue.pop_front() {
+            let pkg = path.last().unwrap();
+            let depth = path.len() as i32 - 1;
+            // generate lines
+            let status = &self.pkg_status.borrow_mut()[&path];
+            if (status.compat.is_changed() || status.latest.is_changed())
+                && (options.flag_packages.is_empty()
+                    || options.flag_packages.contains(&pkg.name().to_string()))
+            {
+                // name version compatible latest kind platform
+                let parent = path.get(path.len() - 2);
+                if let Some(parent) = parent {
+                    let dependency = &self.pkg_deps[parent][pkg];
+                    let label = if self.workspace_mode
+                        || parent == &self.workspace.current()?.package_id()
+                    {
+                        pkg.name().to_string()
+                    } else {
+                        format!("{}->{}", self.pkgs[parent].name(), pkg.name())
+                    };
+
+                    let line = Metadata {
+                        name: label,
+                        project: pkg.version().to_string(),
+                        compat: status.compat.to_string(),
+                        latest: status.latest.to_string(),
+                        kind: format!("{:?}", dependency.kind()).to_string(),
```

Comment:
That effectively relies on debug output. Rust doesn't guarantee any particular format, so this could cause trouble in the future. I suggest using a match.